### PR TITLE
feat: docs update + project structure

### PR DESCRIPTION
**Pull request by zifeo**

### CODE REVIEW

File: website/docs/reference/types/functions.mdx [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+## Effects
+
+Effects are a property of materializers and help categorization what happens to data when it gets transformed. Although they are similar to REST verbs and SQL statements, there is no direct one-to-one mapping.
+
+| Effects  | REST verbs | SQL statements       |
+| -------- | ---------- | -------------------- |
+| `none`   | GET        | `SELECT`             |
+| `create` | POST       | `INSERT`             |
+| `update` | PUT/PATCH  | `UPDATE`             |
+| `upsert` | PUT        | `INSERT ON CONFLICT` |
```
Comment (Natoandro) [inline website/docs/reference/types/functions.mdx:12]: I think we removed the `upsert` effect.

File: typegate/tests/e2e/templates/templates_test.ts [hunk 2]
```diff
@@ -29,9 +29,10 @@ const modifiers: Record<string, (dir: string) => Promise<void> | void> = {
       );
     }
   },
+  "node": () => {},
```
Comment (zifeo) [inline typegate/tests/e2e/templates/templates_test.ts:32]: You have here the ability to replace the import to match to corresponding one in the project (e.g. we want to test on current commit, not the previous published version): see the function above for Deno. For node, you will have to edit package.json and make sure https://github.com/metatypedev/metatype/blob/main/dev/publish.ts is run to create the package.

File: dev/lock.yml [hunk 3]
```diff
@@ -51,9 +51,9 @@ dev:
     examples/**/pyproject.toml:
       (typegraph = ").+("): METATYPE_VERSION
     examples/**/package.json:
-      '(\s*"@typegraph\/sdk"\s*:\s*"\^).+(",?)': METATYPE_VERSION
+      '(\s*"@typegraph\/sdk"\s*:\s*"\^).+(",?)': PUBLISHED_VERSION
     examples/**/*.ts:
-      '(import\s+.+\s+from "npm:@typegraph\/sdk@)[^\/]+(\/.+";)': METATYPE_VERSION
+      '(import\s+.+\s+from "npm:@typegraph\/sdk@)[^\/]+(\/.+";)': PUBLISHED_VERSION
```
Comment (zifeo) [inline dev/lock.yml:56]: see below

**Review by zifeo — COMMENTED** [review]
looks almost good, congrats!

**Comment by Yohe-Am** [comment]
Oof. I thought the "Squash and merge" button would wait for reviews but looks like that behavior is only for protected branches. I guess I'll address any feedback on top of this branch.

**Comment by Yohe-Am** [comment]
A second tutorial that focuses on the GraphQl and HTTP runtimes showcasing how easy it is to consume third-party APIs and more importantly, how to have multiple typegraphs work together....might be a good idea.

**Comment by Yohe-Am** [comment]
Note: when locally building the website, you might run into strange `EPIPE` issues. It only shows up intermittently and I believe it's something introduced along the upgrade to docusaurus 3.0.0. I wasn't able to figure out what was the issue, could be something with my local env. Just wanted to give a word of warning.

That aside, this is good to go.

**Pull request merged**